cmd/clusterctl/cmd/delete.go

```diff
@@ -127,14 +130,15 @@ func runDelete() error {
 		(len(dd.controlPlaneProviders) > 0) ||
 		(len(dd.infrastructureProviders) > 0) ||
 		(len(dd.ipamProviders) > 0) ||
-		(len(dd.runtimeExtensionProviders) > 0)
+		(len(dd.runtimeExtensionProviders) > 0) ||
+		(len(dd.addonProviders) > 0)
 
 	if dd.deleteAll && hasProviderNames {
-		return errors.New("The --all flag can't be used in combination with --core, --bootstrap, --control-plane, --infrastructure, --ipam, --extension")
+		return errors.New("The --all flag can't be used in combination with --core, --bootstrap, --control-plane, --infrastructure, --ipam, --extension, --add-on")
```

Comment:
```suggestion
		return errors.New("The --all flag can't be used in combination with --core, --bootstrap, --control-plane, --infrastructure, --ipam, --extension, --addon")
```
to match the flag definition. same below